src/layouts/MainLayout.vue

```diff
@@ -250,38 +303,101 @@ export default {
       }
       this.last_distribution = last_distribution
     },
-    async update_eth_account () {
-      let account = await ethereum.from_provider(window.ethereum)
+    async update_eth_account (provider) {
+      let account = await ethereum.from_provider(provider)
       this.$store.commit('set_account', account)
     },
-    async web3Connect () {
+    async web3Connect (wallet) {
       let provider = null
-
-      if (window.ethereum) {
-        await window.ethereum.enable()
-
-        provider = new ethers.providers.Web3Provider(window.ethereum)
-      } else if (window.web3) {
-        provider = new ethers.providers.Web3Provider(window.web3.currentProvider)
-      } else {
-        this.$q.notify({
-          type: 'negative',
-          message: 'Non-Ethereum browser detected. You should consider trying MetaMask!'
-        })
-        console.warn('Non-Ethereum browser detected. You should consider trying MetaMask!')
-        return
+      let web3Provider = null
+      if (wallet === 'walletconnect') {
+        try {
+          const wc = new WalletConnectProvider({
+            infuraId: this.infura_key,
+            qrcodeModalOptions: {
+              mobileLinks: [
+                'metamask',
+                'trust'
+              ]
+            }
+          })
+          await wc.enable()
+          provider = new providers.Web3Provider(wc)
+          web3Provider = wc
+          this.web3ConnectModal = false
+        } catch (err) {
+          this.$q.notify({
+            type: 'negative',
+            message: err.message
+          })
+        }
+      } else if (wallet === 'metamask') {
+        if (window.ethereum) {
+          await window.ethereum.enable()
+
+          provider = new ethers.providers.Web3Provider(window.ethereum)
+          web3Provider = window.ethereum
+          this.web3ConnectModal = false
+        } else if (window.web3) {
+          provider = new ethers.providers.Web3Provider(window.web3.currentProvider)
+          web3Provider = window.web3
+          this.web3ConnectModal = false
+        } else {
+          this.$q.notify({
+            type: 'negative',
+            message: 'Non-Ethereum browser detected. You should consider trying MetaMask!'
+          })
+          console.warn('Non-Ethereum browser detected. You should consider trying MetaMask!')
+          return
+        }
+      } else if (wallet == 'phantom') {
+        const isPhantomInstalled = window.solana && window.solana.isPhantom
+        if (isPhantomInstalled) {
```

Comment:
No feedback to the user if there is no Phantom extension installed (`$q.notify` an error to the user could be an option). Furthermore the function execution is not stopped if it's the case; provider is still `undefined` leading to an error from line 374